src/typescript/frontend/src/components/charts/PrivateChart.tsx

```diff
@@ -111,9 +116,27 @@ export const Chart = (props: ChartContainerProps) => {
         onResultReadyCallback(symbols);
       },
       resolveSymbol: async (symbolName, onSymbolResolvedCallback, _onErrorCallback) => {
+        // We pass parameters in the symbolName to prevent re-rendering.
+        // This seems to be the only way to seamlessly change some properties such as has_empty_bars without re-creating the whole chart.
+        // Normally, we would add the state showEmptyCandles in the dependency array to re-create the whole datafeed and the chart.
+        // But this causes the re-creation of the whole chart, and briefly breaks the view (for ex: the custom button gets removed and re-added).
+        // This solution allows to call chart.setSymbol with custom params to update the chart seamlessly.
+        // Uses similar syntax to url query params.
```

Comment:
```suggestion
        // We pass parameters in the symbolName to prevent re-rendering.
        // This seems to be the only way to seamlessly change some properties such as has_empty_bars without re-creating the whole chart.
        // Normally, we would add the state showEmptyCandles in the dependency array to re-create the whole datafeed and the chart.
        // But this causes the re-creation of the whole chart, and briefly breaks the view (for ex: the custom button gets removed and re-added).
        // This solution allows to call chart.setSymbol with custom params to update the chart seamlessly.
        // Uses similar syntax to url query params.
```

When you put this into the other function, at the top of the jsdoc comment for it, just state what it does (passes data into the symbol info, and we remove it later) and then explain why below that